doc/custom_nonconformity_score_proof.ipynb

```diff
@@ -0,0 +1,407 @@
+    "$$\\begin{align}\n",
+    "R_{i,j} &= f(Y_{i}, \\hat{\\mu}_{-(i,j)}(X_{i})) \\\\\n",
+    "R_{i}^{LOO} &= R_{i,n+1}\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## A contruction\n",
+    "There are two variants:\n",
+    "- the symmetrical case when the conformity scores are assumed to be symmetrical around 0. In this case, the absolute values of the conformity scores are used to construct the A matrix.\n",
+    "- otherwise, the signed conformity scores are kept as they are.\n",
+    "\n",
+    "### Symmetrical case\n",
+    "Like in the paper, the A matrix is defined as below:\n",
+    "$$A_{i,j} = \\mathbb{1} \\{ \\lvert R_{i,j} \\rvert > \\lvert R_{j,i} \\rvert \\}$$\n",
+    "\n",
+    "### Asymmetrical case\n",
+    "The two following definitions of matrix A are considered: \n",
+    "$$\\begin{align}\n",
+    "A^{+}_{i,j} &= \\mathbb{1} \\{R_{i,j} > R_{j,i}\\} \\\\\n",
+    "A^{-}_{i,j} &= \\mathbb{1} \\{R_{i,j} < R_{j,i}\\} = J - A^{+}\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Strange points\n",
+    "Like previously, the definition of strange points depends on the case considered.\n",
+    "\n",
+    "### Symmetrical case\n",
+    "Like in the paper, there is a single set $S(A)=\\{i \\in \\{1,...,n+1\\}:A_{i.}\\geq(1-\\alpha)(n+1)\\}$.\n",
+    "\n",
+    "### Asymmetrical case\n",
+    "Instead of the single $S(A)$ set, two cases should be considered: $A^{+}$ and $A^{-}$. Keeping the same definition of the set $S$ as the paper, we have:\n",
+    "- $S_{+}=S(A^{+})=\\{i \\in \\{1,...,n+1\\}:A^{+}_{i.}\\geq(1-\\frac{\\alpha}{2})(n+1)\\}$ for \"data points with unusually large [positive] residuals\", and\n",
+    "- $S_{-}=S(A^{-})=\\{i \\in \\{1,...,n+1\\}:A^{-}_{i.}\\geq(1-\\frac{\\alpha}{2})(n+1) \\iff (n-A^{+}_{i.})\\geq(1-\\frac{\\alpha}{2})(n+1)\\}$  for \"data points with unusually large [negative] residuals\".\n",
+    "\n",
+    "## Step 1 and 2 are unchanged\n",
+    "Changing $R$ and $A$, does not change the results of the steps 1 and 2. From step 1, we have that, in the symetrical case $|S(A)| \\leq 2 \\alpha (n+1)$. And in the asymmetrical case $|S_{+}| \\leq \\alpha (n+1)$ and $|S_{-}| \\leq \\alpha (n+1)$ since the new matrices have the same property as from the paper. From step 2, we have that the probability of the test point $n+1$ being strange inside $S_{+}$ is bounded by $\\alpha$ and the probability of the test point $n+1$ being strange inside $S_{-}$ is also bounded by $\\alpha$ for the asymmetrical case (and the probability of the test point $n+1$ being strange inside $S(A)$ is bounded by $2\\alpha$ for the symmetrical case).\n",
+    "\n",
+    "## Step 3\n",
+    "\n",
+    "### Symmetrical case\n",
+    "\n",
+    "Similarly to the theorem proof, let's suppose that $Y_{n+1} \\notin \\hat{C}_{n,\\alpha}^{jackknife+}(X_{n+1})$. This means either:\n",
+    "$$ Y_{n+1} > \\hat{q}_{n,\\alpha}^{+}\\{ g( \\lvert R_{i}^{LOO} \\rvert , \\hat{\\mu}_{-i}(X_{n+1})) \\}$$\n",
```

Comment:
theorem proof -> theorem's proof
Specify what is \\hat{q}_{n,\\alpha}^{+}

Reply:
What are the differences between these proves and the paper's ones?
It is worth to have these proves in the notebook?

Reply:
- Done
- I added definitions of \hat{q} and of the intervals depending on the considered case.
- The proves in the notebook are more general. Any pair of functions that verify the conditions can be used to construct prediction intervals. The non conformity scores can differ from the absolute value of the residuals.
- If the question is:
  - why a notebook: the goal is not to have the proves in the notebook. It was agreed that I would first write them in a notebook before someone from Quantmetry copies the text into an appropriate file (.rst?).
  - why these proves: after my presentation, I was asked to write the proves that guarantee the coverage for the new non conformity scores.